exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -269,6 +274,18 @@ func (m *AzureManagedMachinePool) validateMaxPods() error {
 	return nil
 }
 
+func (m *AzureManagedMachinePool) validateOSType() error {
+	if m.Spec.OSType != nil && *m.Spec.OSType == azure.WindowsOS {
+		if m.Spec.Mode != string(NodePoolModeUser) {
+			return field.Forbidden(
+				field.NewPath("Spec", "OSType"),
+				"Windows OSType requires AzureManagedMachinePool mode User")
```

Comment:
I have a slight preference for expressing "System node pool must have OSType 'Linux'" or something similar. And so I'd probably invert the validation logic:

- if mode == system
  - if ostype != linux
    - then return error

Reply:
fixed